lib/rubocop/cop/style/map_into_array.rb

```diff
@@ -53,12 +53,17 @@ class MapIntoArray < Base
 
         MSG = 'Use `%<new_method_name>s` instead of `each` to map elements into an array.'
 
+        # @!method suitable_element?(node)
+        def_node_matcher :suitable_element?, <<-PATTERN
+          !{splat forwarded-restarg forwarded-args (block-pass nil?)}
```

Comment:
`map { |x| ... }`, `map { |x| * }`, `map { |x| & }` aren't valid Ruby syntax, so we don't want to correct this type of forwarding